go/interpreter/lfvm/gas.go

```diff
@@ -316,131 +310,71 @@ func callGas(availableGas, base tosca.Gas, callCost *uint256.Int) tosca.Gas {
 	return tosca.Gas(callCost.Uint64())
 }
 
-// Computes the costs for an SSTORE operation
-func gasSStore(c *context) (tosca.Gas, error) {
-	return gasSStoreEIP2200(c)
-}
+func getDynamicCostsForSstore(
+	revision tosca.Revision,
+	storageStatus tosca.StorageStatus,
+) tosca.Gas {
+	isBerlinOrNewer := revision >= tosca.R09_Berlin
 
-//  0. If *gasleft* is less than or equal to 2300, fail the current call.
-//  1. If current value equals new value (this is a no-op), SLOAD_GAS is deducted.
-//  2. If current value does not equal new value:
-//     2.1. If original value equals current value (this storage slot has not been changed by the current execution context):
-//     2.1.1. If original value is 0, SSTORE_SET_GAS (20K) gas is deducted.
-//     2.1.2. Otherwise, SSTORE_RESET_GAS gas is deducted. If new value is 0, add SSTORE_CLEARS_SCHEDULE to refund counter.
-//     2.2. If original value does not equal current value (this storage slot is dirty), SLOAD_GAS gas is deducted. Apply both of the following clauses:
-//     2.2.1. If original value is not 0:
-//     2.2.1.1. If current value is 0 (also means that new value is not 0), subtract SSTORE_CLEARS_SCHEDULE gas from refund counter.
-//     2.2.1.2. If new value is 0 (also means that current value is not 0), add SSTORE_CLEARS_SCHEDULE gas to refund counter.
-//     2.2.2. If original value equals new value (this storage slot is reset):
-//     2.2.2.1. If original value is 0, add SSTORE_SET_GAS - SLOAD_GAS to refund counter.
-//     2.2.2.2. Otherwise, add SSTORE_RESET_GAS - SLOAD_GAS gas to refund counter.
-func gasSStoreEIP2200(c *context) (tosca.Gas, error) {
-	// If we fail the minimum gas availability invariant, fail (0)
-	if c.gas <= SstoreSentryGasEIP2200 {
-		c.status = statusOutOfGas
-		return 0, errNotEnoughGasReentrancy
+	cost := tosca.Gas(800)
```

Comment:
this might be a personal opinion but having `cost` be assigned named variables with significant values (such as `SstoreSetGasEIP2200` or whatever the case would be), seems a lot more significant than just a constant number without any background or reason as for why is that number chosen and not another one. I know this numbers have been calculated with evm codes or something, but having them in a set of predefined constants was very helpful to understand the different cases when reading the code.

Reply:
I think there is not a lot of value of defining constants that are only to be used at one place. Also, naming is hard and if not done systematically and correctly, adds confusion. As it is now, a function like `getRefundForSstore` maps to each potential use case a constant (numeric) value representing its cost, this way documenting the "meaning" of each of those constants.

For instance, if you delete an added memory location in revision London or later, you get a refund of `-4800`. You can see this by simply looking it up in the function, that is (by now) practically a table. Introducing extra constant names like `gasRefundForDeletedAndReAddedStoragePostLondon` would not help readability. Also, if it ever changes in a revision down the road, the `PostLondon` becomes a lie.

Referencing `EIPs` in constants is adding additional confusion as it is very unclear which EIP was active in past or will be active in future Revisions.